js/twister_formatpost.js

```diff
@@ -425,49 +425,74 @@ function htmlFormatMsg(msg, mentions) {
     var i, j, k, str, strEncoded;
     var html = [];
 
     msg = markdown(escapeHtmlEntities(msg),
         '`', 'samp');  // kind of monospace, sequence of chars inside will be escaped from markup
+
     for (i = 0; i < msg.length - 7; i++) {
-        /*if (msg.slice(i, i + 2) === '](') {
+        if (msg.slice(i, i + 2) === '](') {
             // FIXME there can be text with [] inside [] or links with () wee need to handle it too
-            j = getStrStart(msg, i - 1, '[', true, '');
-            if (j < i) {
-                k = getStrEnd(msg, i + 2, ')', true, '');
-                if (k > i + 1) {
-                    html.push($('#external-page-link-template')[0].outerHTML
-                        .replace(/\bid\s*=\s*"[^]*?"+/ig, '')  // $().removeAttr('id')
-                        //.replace(/\bhref\s*=\s*"[^]*?"+/ig, '')  // $().removeAttr('href')
-                        .replace(/<a\s+/ig, '<a href="' + proxyURL(msg.slice(i + 2, k + 1)) + '" ')  // $().closest('a').attr('href', proxyURL(url))
-                        .replace(/(<a\s+[^]*?>)[^]*?(<\/a>)/ig, '$1'
-                            + unpackHtml(
-                                markdown(markdown(markdown(markdown(msg.slice(j, i),
-                                    '*', 'b'),  // bold
-                                    '~', 'i'),  // italic
-                                    '_', 'u'),  // underlined
-                                    '-', 's')  // striketrough
-                                .replace(/&(?!lt;|gt;)/g, '&amp;')
-                                .replace(/"/g, '&quot;')
-                                .replace(/'/g, '&apos;')
-                            )
-                            + '$2')  // $().closest('a').text(url)
-                    );
-                    strEncoded = '>' + (html.length - 1).toString() + '<';
-                    msg = msg.slice(0, j - 1) + strEncoded + msg.slice(k + 2);
-                    i = j + strEncoded.length - 1;
+            j = getStrStart(msg, i - 2, '[', true, '');
+            if (j < i - 1) {
+                k = getStrEnd(msg, i + 3, ')', true, whiteSpaces);
+                if (k > i + 2) {
+                    strEncoded = msg.slice(j, i);  // just temporary storage for name of possiible link
+                    for (i += 2; i < k; i++) {
+                        if (whiteSpacesUrl.indexOf(msg[i]) === -1)  // drop whitespaces and ' and "  // apostrophes and quotes to prevent injection of js events
+                            break;
+                    }
+                    if (i < k) {
+                        // check for injections like [href=(' or ")]('#' or 'javascript' or 'data')[:]  // () and [] here just to explain somehow an idea, it's not about markup syntax
+                        if (msg[i] === '#' || msg.slice(i, i + 10).toLowerCase() === 'javascript'
```

Comment:
by 'explicitly' i mean "blacklisting", i don't think it is safe. i don't know if people aren't going to add another url like 'ecmascript:alert("hello")' in the future...